### PR TITLE
JsLookupResult enhancements

### PR DESCRIPTION
**Pull request by bigjason**
Adds three methods to `JsLookupResult` that make working with lookup results a little more pleasant. These are modeled loosely on how `scala.Option` works. While it is true that `isEmpty` and `isDefined` can be accessed by calling `result.toOption.isDefined`, that method does carry a unused object allocation and extra api noise. So I have preferred these methods instead.

1. `isEmpty` = true if the `JsLookupResult` is a `JsUndefined`.
1. `isDefined` = true if the `JsLookupResult` is a `JsDefined`.
1. `orElse` allows alternate lookups to be easily chained if `isEmpty`.

```scala
Welcome to Scala 2.12.0 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_92).
Type in expressions for evaluation. Or try :help.

scala> import play.api.libs.json._
import play.api.libs.json._

scala> val json = Json.obj("a" -> 11, "b" -> "Hi")
json: play.api.libs.json.JsObject = {"a":11,"b":"Hi"}

scala> (json \ "c") orElse (json \ "d") orElse (json \ "b")
res0: play.api.libs.json.JsLookupResult = JsDefined("Hi")
```

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @bigjason,

Thank you for your contribution! We really value the time you've taken to put this together.

We see that you have signed the Lightbend Contributors License Agreement before, however, the CLA has changed since you last signed it.
Please review the new CLA and sign it before we proceed with reviewing this pull request:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by bigjason** [comment]
@typesafehub-validator I have signed the updated CLA.

File: play-json/src/main/scala/play/api/libs/json/JsLookup.scala [hunk 1]
```diff
@@ -105,6 +105,19 @@ sealed trait JsLookupResult extends Any with JsReadable {
...
+  def isEmpty: Boolean = this match {
+    case JsUndefined() => true
+    case JsDefined(_) => false
+  }
+  def isDefined: Boolean = !isEmpty
+
+  /**
+    * If this result is defined return `this`. Otherwise return `alternative`.
+    */
+  def orElse(alternative: => JsLookupResult): JsLookupResult = this match {
```
Comment (gmethvin) [inline play-json/src/main/scala/play/api/libs/json/JsLookup.scala]: `if (isDefined) this else alternative` (so we can keep the matching logic in one place)?

Comment (bigjason) [inline play-json/src/main/scala/play/api/libs/json/JsLookup.scala]: Good idea. Should I change the other duplicated matches as well?

Comment (gmethvin) [inline play-json/src/main/scala/play/api/libs/json/JsLookup.scala]: Which one(s) are you referring to? I think everywhere else we need to extract the value from the `JsDefined` so it's better to use a match.

Comment (bigjason) [inline play-json/src/main/scala/play/api/libs/json/JsLookup.scala]: You are right. I hadn't thought it through all the way. I updated this though.

**Pull request merged**